src/harness/cbsd_sas_interface.py

```diff
@@ -0,0 +1,167 @@
+
+    @abc.abstractmethod
+    def Heartbeat(self, request, ssl_cert=None, ssl_key=None):
+        """SAS-CBSD Heartbeat interface.
+
+        Requests heartbeat for a grant for CBSDs.
+
+        Request and response are both lists of dictionaries. Each dictionary
+        contains all fields of a single request/response.
+
+        Args:
+          request: A dictionary with a single key-value pair where the key is
+            "heartbeatRequest" and the value is a list of individual CBSD
+            heartbeat requests (each of which is itself a dictionary).
+          ssl_cert: Path to SSL cert file, if None, will use default cert file.
+          ssl_key: Path to SSL key file, if None, will use default key file.
+        Returns:
+          A dictionary with a single key-value pair where the key is
+          "heartbeatResponse" and the value is a list of individual CBSD
+          heartbeat responses (each of which is itself a dictionary).
+        """
+        pass
+
+    @abc.abstractmethod
+    def Relinquishment(self, request, ssl_cert=None, ssl_key=None):
+        """SAS-CBSD Relinquishment interface.
+
+        Relinquishes grant for CBSDs.
+
+        Request and response are both lists of dictionaries. Each dictionary
+        contains all fields of a single request/response.
+
+        Args:
+          request: A dictionary with a single key-value pair where the key is
+            "relinquishmentRequest" and the value is a list of individual CBSD
+            relinquishment requests (each of which is itself a dictionary).
+          ssl_cert: Path to SSL cert file, if None, will use default cert file.
+          ssl_key: Path to SSL key file, if None, will use default key file.
+        Returns:
+          A dictionary with a single key-value pair where the key is
+          "relinquishmentResponse" and the value is a list of individual CBSD
+          relinquishment responses (each of which is itself a dictionary).
+        """
+        pass
+
+    @abc.abstractmethod
+    def Deregistration(self, request, ssl_cert=None, ssl_key=None):
+        """SAS-CBSD Deregistration interface.
+
+        Deregisters CBSDs.
+
+        Request and response are both lists of dictionaries. Each dictionary
+        contains all fields of a single request/response.
+
+        Args:
+          request: A dictionary with a single key-value pair where the key is
+            "deregistrationRequest" and the value is a list of individual CBSD
+            deregistration requests (each of which is itself a dictionary).
+          ssl_cert: Path to SSL cert file, if None, will use default cert file.
+          ssl_key: Path to SSL key file, if None, will use default key file.
+        Returns:
+          A dictionary with a single key-value pair where the key is
+          "deregistrationResponse" and the value is a list of individual CBSD
+          deregistration responses (each of which is itself a dictionary).
+        """
+        pass
+
```

Comment:
add SetVertionNumber into this file

basically this file contains all interfaces that test script writer can use